dnsimple/dnsimple.go

```diff
@@ -370,3 +370,11 @@ func addURLQueryOptions(path string, options interface{}) (string, error) {
 
 	return u.String(), nil
 }
+
+// Int64P is a helper routine that allocates a new int64 value
+// to store v and returns a pointer to it.
+func Int64P(v int64) *int64 { return &v }
+
+// StringP is a helper routine that allocates a new string value
+// to store v and returns a pointer to it.
+func StringP(v string) *string { return &v }
```

Comment:
I'm struggling to understand why there should be helpers rather than just creating the pointer inline. Could you provide some insight?

Reply:
Because the only way to reference a pointer is if you have stored the value somewhere, usually in a variable.

But this is not always the case. For instance, the specs demonstrates how you may call the code inlining the string:

https://github.com/dnsimple/dnsimple-go/pull/92/files#diff-20edf994fa1e7a7cc52e13d834a3a4a4R176

Without this helper you could not do it, you cannot obtain the pointer of a string literal.

Reply:
Thank you for the explanation. 👍